src/components/SharedComponents/hooks/fetchUserLocation.js

```diff
@@ -0,0 +1,58 @@
+// @flow
```

Comment:
So far I've been naming all the hooks file names with `fetch` and the actual functions with `use`. I guess I like fetch because it reminds me that we're retrieving data, but I'm certainly not tied to that naming convention.